src/components/posts/view-post/PostPage.tsx

```diff
@@ -109,34 +112,42 @@ export const PostPage: NextPage<PostDetailsProps> = ({ postDetails: initialPost,
       </Section>
     </PageContent>
   </>
-};
+}
 
 PostPage.getInitialProps = async (props): Promise<any> => {
-  const { query: { spaceId, postId }, res } = props;
+  const { query: { spaceId, slug }, res } = props
+
   const subsocial = await getSubsocialApi()
-  const { substrate } = subsocial;
+  const { substrate } = subsocial
   const idOrHandle = spaceId as string
-  const spaceIdFromUrl = await getSpaceId(idOrHandle)
 
-  const postIdFromUrl = new BN(postId as string)
+  const slugStr = slug as string
+  const postIdFromUrl = getPostIdFromSlug(slugStr)
   const replyIds = await substrate.getReplyIdsByPostId(postIdFromUrl)
   const comments = await subsocial.findPublicPostsWithAllDetails([ ...replyIds, postIdFromUrl ])
 
   const [ extPostsData, replies ] = partition(comments, x => x.post.struct.id.eq(postIdFromUrl))
   const extPostData = extPostsData.pop() || await subsocial.findPostWithAllDetails(postIdFromUrl)
 
-  const spaceIdFromPost = unwrapSubstrateId(extPostData?.post.struct.space_id)
-  // If a space id of this post is not equal to the space id/handle from URL,
-  // then redirect to the URL with the space id of this post.
-  if (spaceIdFromPost && spaceIdFromUrl && !spaceIdFromPost.eq(spaceIdFromUrl) && res) {
-    res.writeHead(301, { Location: `/${spaceIdFromPost.toString()}/posts/${postId}` })
+  const spaceIdFromPost = unwrapSubstrateId(extPostData?.post.struct.space_id) as SpaceId
+
+  const currentSpace = { id: spaceIdFromPost, handle: idOrHandle } as unknown as Space
+  const currentPostUrl = spaceUrl({ id: spaceIdFromPost, handle: idOrHandle } as unknown as Space, slugStr)
```

Comment:
Seems like a copy-paste to me. Should it me this?
```suggestion
  const currentSpace = { id: spaceIdFromPost, handle: idOrHandle } as unknown as Space
  const currentPostUrl = spaceUrl(currentSpace, slugStr)
```